ARCHITECTURE.md

```diff
@@ -1,113 +1,265 @@
+The purpose of `Persistent Player` is to store synchronized data about player, such as their name, chosen avatar GUID etc.
 
-* PersistentPlayer: a `NetworkObject` that will not be destroyed between scenes
+This `PlayerAvatar` prefab instance, that is owned by the corresponding connected client, is destroyed by Netcode when a scene load occurs (either to `PostGame` scene, or back to `MainMenu` scene), or through client disconnection.
 
-####CharSelect Scene
 Inside `CharSelect` scene, clients select from 8 possible avatar classes, and that selection is stored inside PersistentPlayer's `NetworkAvatarGuidState`.
 
-####BossRoom Scene
-Inside `BossRoom` scene, `ServerBossRoomState` spawns a `PlayerAvatar` per PersistentPlayer present. This `PlayerAvatar` prefab instance, that is owned by the corresponding connected client, is destroyed by Netcode when a scene load occurs (either to `PostGame` scene, or back to `MainMenu` scene), or through client disconnection.
+Inside `BossRoom` scene, `ServerBossRoomState` spawns a `PlayerAvatar` per PersistentPlayer present.
+
+`ClientAvatarGuidHandler`, a `NetworkBehaviour` component residing on the `PlayerAvatar` prefab instance will fetch the validated avatar GUID from `NetworkAvatarGuidState`, and spawn a local, non-networked graphics GameObject corresponding to the avatar GUID.
+
+### Characters
+
+`ServerCharacter` contains NetworkVariables that store the state of any given character, and the server RPCs. It is responsible for executing or kicking off the server-side logic for the characters, which includes:
+
+- movement and pathfinding via `ServerCharacterMovement` - it uses NavMeshAgent that lives on the server to translate the character’s transform, which is synchronized using NetworkTransform component;
+- player action queueing and execution via `ServerActionPlayer`;
+- AI logic via `AIBrain` (applies to NPCs);
+- Character animations via `ServerAnimationHandler`, which themselves are synchronized using NetworkAnimator;
+
+`ClientCharacter` primarily is a host for the `ClientActionPlayer` class, and it also contains the client RPCs for the character gameplay logic.
+
+### Game data setup
+
+Game data in Boss Room is defined in `ScriptableObjects`.
+
+A singleton class [GameDataSource.cs](Assets/Scripts/Gameplay/GameplayObjects/RuntimeDataContainers/GameDataSource.cs) is responsible for storing all the actions and character classes in the game.
+
+[CharacterClass](Assets/Scripts/Gameplay/Configuration/CharacterClass.cs) is the data representation of a Character, containing such things as its starting stats and a list of Actions that it can perform. This covers both player characters and NPCs alike.
+
+[Action](Assets/Scripts/Gameplay/Configuration/Action.cs) subclasses represent discrete verbs (like swinging a weapon, or reviving someone), and are substantially data driven.
+
+### Action System
```

Comment:
need to add a note that users will need to build their own action system for game designers if they want something extensible that allows game design emergence. Boss Room's action system is just for boss room's purpose.